conans/search/search.py

```diff
@@ -136,7 +136,10 @@ def _get_local_infos_min(paths, reference):
             if not os.path.exists(info_path):
                 raise NotFoundException("")
             conan_info_content = load(info_path)
-            conan_vars_info = ConanInfo.loads(conan_info_content).serialize_min()
+            info = ConanInfo.loads(conan_info_content)
+            if reference.revision and info.recipe_revision and info.recipe_revision != reference.revision:
+                continue
```

Comment:
This silent skipping of existing binaries could be confusing, maybe warn or something.

Reply:
Probably we should avoid discarding packages. The "remove" is using this search and it feels weird a remove not removing some packages. Should we only block the upload in the client and let the user mix in the local cache?

Reply:
Maybe it is ok. It is only discarded when the user specifies a revision in the search/remove command. So it is not so weird to skip some packages. About the remove, it works ok this way.